### PR TITLE
Print api response in logs for 4xx errors

### PR DESCRIPTION
**Pull request by meain**
## Description

Print api response in logs for 4xx errors from GC. More info for things like 400 Bad Request could be useful.

## Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No 

## Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #<issue>

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/graph/service.go [hunk 1]
```diff
@@ -307,9 +307,16 @@ func (handler *LoggingMiddleware) Intercept(
...
+		} else if resp.StatusCode/100 == 4 {
+			respDump, _ := httputil.DumpResponse(resp, true)
+			logger.Ctx(ctx).Infow(
+				"graph api error",
+				"status", resp.Status,
+				"method", req.Method,
+				"url", req.URL,
+				"response", string(respDump),
+			)
+		} else if resp.StatusCode/100 != 2 {
```
Comment (ryanfkeepers) [inline src/internal/connector/graph/service.go:319]: What is this solving that the above conditional (toggling all api logging) doesn't?

File: src/internal/connector/graph/service.go [hunk 2]
```diff
@@ -307,9 +307,16 @@ func (handler *LoggingMiddleware) Intercept(
...
-
-		if resp.StatusCode != http.StatusTooManyRequests && (resp.StatusCode/100) != 2 {
+		} else if resp.StatusCode/100 == 4 {
+			respDump, _ := httputil.DumpResponse(resp, true)
+			logger.Ctx(ctx).Infow(
+				"graph api error",
+				"status", resp.Status,
+				"method", req.Method,
+				"url", req.URL,
+				"response", string(respDump),
```
Comment (ryanfkeepers) [inline src/internal/connector/graph/service.go:317]: Preference is to go the simpler route and dump the response on all non-2xx, non-throttling responses.

**Review by ryanfkeepers — APPROVED** [review]
Approving to unblock, but I don't see the value here when we already have the additional logging flags for getting the same information.  Keeping this to a flag helps maintain less chatty logs for standard runs.

File: src/internal/connector/graph/service.go (hunk 1 above)
Comment (meain) [reply src/internal/connector/graph/service.go:319]: This was done primarily to make sure we get some extra information about why requests failed in ark without having to enable debug logging which can be really noisy. The primary status code that I think logging resopnse could help is just 400 (not necessarily 4xx) as having the response might help us figure out what is causing a failure like in the case of possible "restricted directories". Let me know what you think.

Comment (ryanfkeepers) [reply src/internal/connector/graph/service.go:319]: mm, I still think we'll get all this information out of the new structured errors.  But don't let me stop you.

If we're only looking for 400, we should explicitly catch that; not all 4xx.
```
else if resp.Status == http.StatusBadRequest {
```

**Pull request merged**